src/pd/pd.rs

```diff
@@ -825,13 +826,15 @@ fn send_admin_request(
 }
 
 /// Sends merge fail message to gc merge source.
-fn send_merge_fail(ch: SendCh, source_region_id: u64, target: metapb::Peer) {
+fn send_merge_fail(router: &RaftRouter, source_region_id: u64, target: metapb::Peer) {
     let target_id = target.get_id();
-    if let Err(e) = ch.send(Msg::PeerMsg(PeerMsg::MergeResult {
-        region_id: source_region_id,
-        target,
-        stale: true,
-    })) {
+    if let Err(e) = router.send(
+        source_region_id,
+        PeerMsg::CasualMessage(CasualMessage::MergeResult {
```

Comment:
`MergeResult` is not casual. For the merge isolation recovery mechanism, it relies on `MergeResult` to destroy overlapping source peer.

Reply:
If it's lost, it will be resent again.

Reply:
where to resent?

Reply:
seems only resent in `try_send`, but it is still possible that `MergeResult` fails to send due to channel full.